src/shm_sub.c

```diff
@@ -2292,6 +2310,20 @@ sr_shmsub_rpc_internal_call_callback(sr_conn_ctx_t *conn, const struct lyd_node
             goto cleanup;
         }
     }
+    ly_set_free(set, NULL);
+
+    /* get datastores which should be used for reset */
+    if ((err_info = sr_lyd_find_xpath(input, "datastores/datastore", &set))) {
+        goto cleanup;
+    }
+    for (i = 0; i < set->count; ++i) {
+        if (!str2ds(lyd_get_value(set->dnodes[i]), &ds)) {
+            datastores |= (1 << ds);
```

Comment:
Please no hacks like these, some more readable code instead. For example, `int reset_ds[2]` with either `0` or `1` on `SR_DS_*` index if the specific datastore should be reset or not.